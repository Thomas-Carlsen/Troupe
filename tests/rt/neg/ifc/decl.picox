let val x = 42 raisedTo {alice,bob}
    val z = {alice}
    val aliceAuthority = attenuate (authority, {alice})
    val bobAuthority = attenuate(authority, {bob})    
    val y = declassify (x, aliceAuthority, z)
in ()
end
