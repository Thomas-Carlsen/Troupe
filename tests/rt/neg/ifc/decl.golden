2019-03-03T00:32:37.103Z [RTM] [32minfo[39m: Skipping network creation. Observe that all external IO operations will yield a runtime error.
[31mRuntime error in thread c0ee4bb4-0c80-4fb9-a63c-f9664c1715e0@{}[39m
[31m>> Not enough authority for declassification[39m
[31m | level of the data: {alice,bob}[39m
[31m | level of the authority: {alice}[39m
[31m | target level of the declassification: {alice}[39m
